inbox/models/category.py

```diff
@@ -39,21 +48,29 @@ def API_OBJECT_NAME(self):
 
     # STOPSHIP(emfree): need to index properly for API filtering performance.
     name = Column(String(MAX_INDEXABLE_LENGTH), nullable=False, default='')
-    display_name = Column(String(MAX_INDEXABLE_LENGTH,
-                                 collation='utf8mb4_bin'), nullable=False)
+    display_name = Column(CategoryNameType(MAX_INDEXABLE_LENGTH,
+                          collation='utf8mb4_bin'), nullable=False)
 
     type_ = Column(Enum('folder', 'label'), nullable=False, default='folder')
 
     @validates('display_name')
-    def sanitize_display_name(self, key, display_name):
-        if self.type_ == 'label':
-            display_name = unicode(display_name)
-        display_name = display_name.rstrip()
-        if len(display_name) > MAX_INDEXABLE_LENGTH:
-            log.warning('Truncating category name',
-                        type_=self.type_, original=display_name)
-            display_name = display_name[:MAX_INDEXABLE_LENGTH]
-        return display_name
+    def validate_display_name(self, key, display_name):
+        sanitized_name = Category.sanitize_name(display_name)
+        if sanitized_name != display_name:
+            log.warning("Truncating category display_name", type_=self.type_,
+                        account_id=self.account_id, original=display_name)
+        return sanitized_name
+
+    @classmethod
+    def sanitize_name(cls, name):
+        # g_label may not have unicode type (in particular for a numeric
+        # label, e.g. '42'), so coerce to unicode.
+        name = unicode(name)
```

Comment:
I think this should be fine. I am in favour of having all of our internal processing use unicode objects and only convert strings to bytestrings when needed to transmit them to the outside world.